src/main/java/org/opensearch/knn/index/SpaceType.java

```diff
@@ -172,6 +199,12 @@ public static Set<String> getValues() {
 
     public static SpaceType getSpace(String spaceTypeName) {
         for (SpaceType currentSpaceType : SpaceType.values()) {
+            // UNDEFINED space type is a temporary value to be used only until we set proper default value when
+            // space type is not provided by user
+            // Therefore, we do not allow converting space type name to UNDEFINED space type
+            if (SpaceType.UNDEFINED == currentSpaceType) {
+                continue;
+            }
```

Comment:
how will someone validate that space type is UNDEFINED and they need to set the space type?